mlir-sycl/test/Transforms/constant-propagation.mlir

```diff
@@ -1170,3 +1184,194 @@ llvm.func internal @foo_constant_constant_global_local_size(
+    gpu.return
+  }
+
+// COM: No access range needed, constant memory range and default offset.
+
+// CHECK-LABEL:     gpu.func @k2(
+// CHECK-SAME:                   %[[VAL_18:.*]]: memref<?xi32, 1>, %[[VAL_19:.*]]: memref<?x!sycl_range_1_>, %[[VAL_20:.*]]: memref<?x!sycl_range_1_>, %[[VAL_21:.*]]: memref<?x!sycl_id_1_>) kernel {
+// CHECK-NEXT:        %[[VAL_22:.*]] = arith.constant 512 : index
+// CHECK-NEXT:        %[[VAL_23:.*]] = sycl.range.constructor(%[[VAL_22]]) : (index) -> memref<1x!sycl_range_1_>
+// CHECK-NEXT:        %[[VAL_24:.*]] = memref.cast %[[VAL_23]] : memref<1x!sycl_range_1_> to memref<?x!sycl_range_1_>
+// CHECK-NEXT:        %[[VAL_25:.*]] = sycl.id.constructor() : () -> memref<1x!sycl_id_1_>
+// CHECK-NEXT:        %[[VAL_26:.*]] = memref.cast %[[VAL_25]] : memref<1x!sycl_id_1_> to memref<?x!sycl_id_1_>
+// CHECK-NEXT:        %[[VAL_27:.*]] = memref.alloca() : memref<1x!sycl_accessor_1_i32_w_gb2>
+// CHECK-NEXT:        func.call @init(%[[VAL_27]], %[[VAL_18]], %[[VAL_24]], %[[VAL_24]], %[[VAL_26]]) : (memref<1x!sycl_accessor_1_i32_w_gb2>, memref<?xi32, 1>, memref<?x!sycl_range_1_>, memref<?x!sycl_range_1_>, memref<?x!sycl_id_1_>) -> ()
+// CHECK-NEXT:        gpu.return
+// CHECK-NEXT:      }
+  gpu.func @k2(%ptr: memref<?xi32, 1>,
+               %accRange: memref<?x!sycl_range_1_>,
+               %memRange: memref<?x!sycl_range_1_>,
+               %offset: memref<?x!sycl_id_1_>) kernel {
+    %acc = memref.alloca() : memref<1x!sycl_accessor_1_i32_w_gb>
+    func.call @init(%acc, %ptr, %accRange, %memRange, %offset)
+        : (memref<1x!sycl_accessor_1_i32_w_gb>, memref<?xi32, 1>,
+           memref<?x!sycl_range_1_>, memref<?x!sycl_range_1_>,
+           memref<?x!sycl_id_1_>) -> ()
+    gpu.return
+  }
+
+// COM: Not enough info
+
+// CHECK-LABEL:     gpu.func @k3(
+// CHECK-SAME:                   %[[VAL_28:.*]]: memref<?xi32, 1>, %[[VAL_29:.*]]: memref<?x!sycl_range_1_>, %[[VAL_30:.*]]: memref<?x!sycl_range_1_>, %[[VAL_31:.*]]: memref<?x!sycl_id_1_>) kernel {
+// CHECK-NEXT:        %[[VAL_32:.*]] = memref.alloca() : memref<1x!sycl_accessor_1_i32_w_gb2>
+// CHECK-NEXT:        func.call @init(%[[VAL_32]], %[[VAL_28]], %[[VAL_29]], %[[VAL_30]], %[[VAL_31]]) : (memref<1x!sycl_accessor_1_i32_w_gb2>, memref<?xi32, 1>, memref<?x!sycl_range_1_>, memref<?x!sycl_range_1_>, memref<?x!sycl_id_1_>) -> ()
+// CHECK-NEXT:        gpu.return
+// CHECK-NEXT:      }
+  gpu.func @k3(%ptr: memref<?xi32, 1>,
+               %accRange: memref<?x!sycl_range_1_>,
+               %memRange: memref<?x!sycl_range_1_>,
+               %offset: memref<?x!sycl_id_1_>) kernel {
+    %acc = memref.alloca() : memref<1x!sycl_accessor_1_i32_w_gb>
+    func.call @init(%acc, %ptr, %accRange, %memRange, %offset)
+        : (memref<1x!sycl_accessor_1_i32_w_gb>, memref<?xi32, 1>,
+           memref<?x!sycl_range_1_>, memref<?x!sycl_range_1_>,
+           memref<?x!sycl_id_1_>) -> ()
+    gpu.return
+  }
+}
+
+llvm.func internal @constant_offset_and_range(
```

Comment:
Nit: Can we move these functions and group each pair of host function with the device kernel?